src/GloryDuckTables2019Lkl.cc

```diff
@@ -207,51 +207,73 @@ Int_t GloryDuckTables2019Lkl::ReadGloryDuckInputData(TString filename)
   else
     cout << "GloryDuckTables2019Lkl::ReadGloryDuckInputData (" << GetName() << ") Message: Reading masses and <sigmav> values from file " << filename << endl;
 
-  vector<Double_t> vsigma ;    // vector with <sv> values
-  vector<Double_t> vlkl ;      // vecot with -2logL values
-
-  // get first line of the file (with <sv> values)
-  string sigmavLine;
-  getline(ff,sigmavLine);
+  // get the number of lines of the file
+  Int_t numRows = 0;
+  string line;
+  while (getline(ff, line))
+    numRows++;
+  ff.clear();
+  ff.seekg(0, ios::beg);
+
+  vector<Double_t> sigmav ;                   // vector with <sv> values
+  vector<Double_t> mass ;                     // vector with mass values
+  vector<vector<Double_t> > vlkl2D(numRows-1);// 2D vector with -2logL values
+  vector<Double_t> vlkl;                      // vector with -2logL values
+
+  // get first line of the file (with mass values)
+  string massLine;
+  getline(ff,massLine);
 
   // read first line of the file
-  std::istringstream row(sigmavLine);
+  std::istringstream first_line(massLine);
   Double_t field;
-  while (row>> field)
+  while (first_line >> field)
     {
+      if(fNsvVals != 0)
+        mass.push_back(field);
       fNsvVals++;
-      vsigma.push_back(field);
     }
 
-  Double_t readingMass = 0.;
+  Double_t readingSigmav = 0.;
   Double_t readingLkl = 0.;
-  Int_t counter = 0;
+  Int_t col = 0;
+  Int_t row = 0;
 
-  // get mass (first column)
-  while(ff >> readingMass)
+  // get <sv> (first column)
+  while(ff >> readingSigmav)
     {
-      // get rest of the line (<sv> values)
+      // store the <sv> values
+      sigmav.push_back(readingSigmav);
+
+      // get rest of the line
       string LklLine;
       getline(ff,LklLine);
 
       // read rest of the line
+      vlkl2D[col] = vector<Double_t>(mass.size());
+      row = 0;
       std::istringstream line(LklLine);
       while (line >> readingLkl)
         {
-          vlkl.push_back(readingLkl);
+          vlkl2D[col][row] = readingLkl;
+          row++;
         }
-
-      // add new parabola
-      CreateAndAddNewParabola(readingMass,vsigma.size(),vsigma.data(),vlkl.data());
-
-      // check if line was complete or not
-      if(vlkl.size() != vsigma.size())
+      // check if line was complete or not HERE -1 LATER
+      if(vlkl2D[col].size() != mass.size())
         {
-          cout << "You have a different line size length for line " << counter+1 << " ! Number of sigmav values = " << vsigma.size() << " while number of likelihood values = " << vlkl.size() << endl; 
+          cout << "You have a different line size length for line " << col+1 << " ! Number of mass values = " << mass.size() << " while number of likelihood values = " << vlkl2D[col].size() << endl;
         }
-      counter++;
-      vlkl.clear();
-   }
+      col++;
+    }
+
+  // add new parabolas
+  for (row = 0; row < fNsvVals-1; row++) {
```

Comment:
I'm a bit lost here because of my own comment above but I believe (following the logic of the file) it should be something like
for (col = 0; col < fNMasses; col++) {
  for (row = 0; row < fNsvVals; row++) {
so that you create as many parabolas as there is masses and the parabolas are of length the number of sv values.

Reply:
I think the code is doing what you mention here, but the names are wrong!